### PR TITLE
[CH-107] Support Native RowToColumnar

### PR DESCRIPTION
**Pull request by binmahone**
Changelog category (leave one):
- New Feature
- Improvement
- Bug Fix (user-visible misbehaviour in official stable or prestable release)
- Performance Improvement
- Backward Incompatible Change
- Build/Testing/Packaging Improvement
- Documentation (changelog entry is not required)
- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

> Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: utils/local-engine/Parser/SparkRowToCHColumn.cpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            columns[i]->insertData(spark_row_reader.getRawDataForFixedNumber(i), sizeof(uint32_t));
+        }
+        else if (which.isString())
+        {
+            StringRef data = spark_row_reader.getString(i);
+            columns[i]->insertData(data.data, data.size);
+        }
+        else
+        {
+            throw std::runtime_error("doesn't support type " + String(magic_enum::enum_name(columns[i]->getDataType())));
```
Comment (liuneng1994) [inline utils/local-engine/Parser/SparkRowToCHColumn.cpp:92]: 优先使用clickhouse的Exception

File: utils/local-engine/Parser/SparkRowToCHColumn.h [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+
+namespace ErrorCodes
+{
+    extern const int UNKNOWN_TYPE;
+}
+
+struct SparkRowToCHColumnHelper
+{
+    SparkRowToCHColumnHelper(vector<string> names, vector<string> types, vector<bool> isNullables)
```
Comment (liuneng1994) [inline utils/local-engine/Parser/SparkRowToCHColumn.h:27]: 这里是不是可以传引用减少对象拷贝

File: utils/local-engine/Parser/SparkRowToCHColumn.h [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+            return std::make_shared<DataTypeNullable>(nested_type);
+        }
+        else
+        {
+            return nested_type;
+        }
+    }
+
+    //parse Spark type name to CH DataType
+    DataTypePtr parseType(const string type, const bool isNullable)
```
Comment (liuneng1994) [inline utils/local-engine/Parser/SparkRowToCHColumn.h:73]: type可以是引用

File: utils/local-engine/local_engine_jni.cpp [hunk 4]
```diff
@@ -78,6 +78,16 @@ jint JNI_OnLoad(JavaVM * vm, void * reserved)
         = GetMethodID(env, local_engine::SourceFromJavaIter::serialized_record_batch_iterator_class, "hasNext", "()Z");
     local_engine::SourceFromJavaIter::serialized_record_batch_iterator_next
         = GetMethodID(env, local_engine::SourceFromJavaIter::serialized_record_batch_iterator_class, "next", "()[B");
+
+
+    local_engine::SparkRowToCHColumn::spark_row_interator_class
```
Comment (liuneng1994) [inline utils/local-engine/local_engine_jni.cpp:83]: 没看到回收global class的代码

**Review by liuneng1994 — CHANGES_REQUESTED** [review]

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**